service/src/main/java/tds/assessment/repositories/impl/ItemQueryRepositoryImpl.java

```diff
@@ -97,48 +95,37 @@ public ItemQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTemplate) {
                 "AND \n" +
                 "   formItem.isactive = 1";
 
-        return jdbcTemplate.query(itemsSQL, parameters, resultExtractor -> {
-            Map<String, Item> itemsMap = new HashMap<>();
-
-            while (resultExtractor.next()) {
-                Item item = itemsMap.get(resultExtractor.getString("id"));
-                if (item == null) {
-                    item = new Item(resultExtractor.getString("id"));
-                    itemsMap.put(item.getId(), item);
-                }
-
-                item.setSegmentKey(resultExtractor.getString("segmentKey"));
-                item.setItemType(resultExtractor.getString("itemtype"));
-                item.setGroupId(resultExtractor.getString("groupid"));
-                item.setGroupKey(resultExtractor.getString("groupkey"));
-                item.setBlockId((resultExtractor.getString("blockid")));
-                item.setPosition(resultExtractor.getInt("position"));
-                item.setFormPosition(resultExtractor.getInt("formPosition"));
-                item.setFieldTest(resultExtractor.getBoolean("isfieldtest"));
-                item.setRequired(resultExtractor.getBoolean("isrequired"));
-                item.setStrand(resultExtractor.getString("strandname"));
-                item.getFormKeys().add(resultExtractor.getString("formKey"));
-                item.setItemFilePath(resultExtractor.getString("itemFilePath"));
-                item.setStimulusFilePath(resultExtractor.getString("stimulusFilePath"));
-                item.setPrintable(resultExtractor.getBoolean("isprintable"));
-                item.setMimeType(resultExtractor.getString("responseMimeType"));
-                item.setMaxScore(resultExtractor.getInt("maxScore"));
-                item.setContentLevel(resultExtractor.getString("contentLevel"));
-                item.setClientId(resultExtractor.getString("clientId"));
-                item.setNotForScoring(resultExtractor.getBoolean("notForScoring"));
-                item.setbVector(resultExtractor.getString("bVector"));
-                item.setItemResponseTheoryModel(resultExtractor.getString("irtModel"));
-                item.setItemResponseTheoryAParameter((Float) resultExtractor.getObject("irtA"));
-                item.setItemResponseTheoryBParameter(resultExtractor.getString("irtB"));
-                item.setItemResponseTheoryCParameter((Float) resultExtractor.getObject("irtC"));
-                item.setClaims(resultExtractor.getString("clString"));
-                item.setItemKey(resultExtractor.getLong("itemKey"));
-                item.setBankKey(resultExtractor.getLong("bankKey"));
-            }
+        return jdbcTemplate.query(itemsSQL, parameters, (rs, rowNum) -> {
```

Comment:
So this really shouldn't happen much anymore because we're not getting item records across segments which is what was happening prior to the query change above.  That being said it seemed like his had a bit too much logic when trying to collapse the items.  Instead, we'll return all the items and allow the assembler to handle it.